src/main/java/teammates/ui/webapi/GetInstructorsAction.java

```diff
@@ -28,41 +31,124 @@ void checkSpecificAccessControl() throws UnauthorizedAccessException {
-        List<InstructorAttributes> instructorsOfCourse = logic.getInstructorsForCourse(courseId);
-
+        String intentStr = getRequestParamValue(Const.ParamsNames.INTENT);
         InstructorsData data;
 
-        String intentStr = getRequestParamValue(Const.ParamsNames.INTENT);
-        if (intentStr == null) {
-            instructorsOfCourse =
-                    instructorsOfCourse.stream()
-                            .filter(InstructorAttributes::isDisplayedToStudents)
+        if (!isCourseMigrated(courseId)) {
+            List<InstructorAttributes> instructorsOfCourse = logic.getInstructorsForCourse(courseId);
+
+            if (intentStr == null) {
+                data = new InstructorsData();
+                instructorsOfCourse =
+                        instructorsOfCourse.stream()
+                                .filter(InstructorAttributes::isDisplayedToStudents)
+                                .collect(Collectors.toList());
+
+                // To map each InstructorAttributes object to InstructorData format
+                // in order to fill the list in InstructorsData via the setter method.
+                List<InstructorData> instructorDataList = instructorsOfCourse
+                        .stream()
+                        .map(InstructorData::new)
+                        .collect(Collectors.toList());
+
+                data.setInstructors(instructorDataList);
+
+                // hide information
+                data.getInstructors().forEach(i -> {
+                    i.setGoogleId(null);
+                    i.setJoinState(null);
+                    i.setIsDisplayedToStudents(null);
+                    i.setRole(null);
+                });
+            } else if (intentStr.equals(Intent.FULL_DETAIL.toString())) {
+                // get all instructors of a course without information hiding
+                // adds googleId if caller is admin or has the appropriate privilege to modify instructor
+                if (userInfo.isAdmin || logic.getInstructorForGoogleId(courseId, userInfo.getId()).getPrivileges()
+                        .isAllowedForPrivilege(Const.InstructorPermissions.CAN_MODIFY_INSTRUCTOR)) {
+                    data = new InstructorsData();
+                    for (InstructorAttributes instructor : instructorsOfCourse) {
+                        InstructorData instructorData = new InstructorData(instructor);
+                        instructorData.setGoogleId(instructor.getGoogleId());
+                        if (userInfo.isAdmin) {
+                            instructorData.setKey(instructor.getKey());
+                        }
+                        data.getInstructors().add(instructorData);
+                    }
+                } else {
+                    data = new InstructorsData();
+
+                    // To map each InstructorAttributes object to InstructorData format
+                    // in order to fill the list in InstructorsData via the setter method.
```

Comment:
This is definitely not what I meant. This comment is just noise, doesn't add any value whatsoever.

I was referring to explaining why you had to use the setter instead of the constructor.